src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy

```diff
@@ -88,8 +94,23 @@ class GitVersionPlugin implements Plugin<Project> {
             // first to preserve this behavior in cases where this call would fail but native "git" call does not.
             new DescribeCommand(git.getRepository()).call()
 
-            return GitCli.runGitCommand(project.rootDir, "describe", "--tags", "--always", "--first-parent",
-                    "--match=${prefix}*")
+            /*
+             * Mimick 'git describe --tags --always --first-parent --match=${prefix}*' by using rev-list to
+             * support versions of git < 1.8.4
+             */
+            int depth = 0
+            String revList = GitCli.runGitCommand(project.rootDir, "rev-list", "--first-parent", "HEAD")
+            for (String rev : revList.split(System.getProperty("line.separator"))) {
```

Comment:
Rather than a separate `int depth = 0` declaration and this nice java for each loop, what about using an old school `for(int depth=0; depth < revListArray.length; depth++)` to emphasise that depth is just the index of the hash that we're looking at?